app/Metadata/GitHubFunctions.php

```diff
@@ -212,18 +227,52 @@ public function get_behind_text()
 		}
 
 		$count = $this->count_behind();
+
+		/** @var Carbon|null $last_update */
+		$last_update = $this->get_update_age();
+		if (!$last_update) {
+			$last = 'unknown';
+			$end = '';
+		} else {
+			$last = now()->diffInDays($last_update);
+			$end = $last > 0 ? ' days' : '';
+			$last = ($last == 0 && $end = ' hours') ? now()->diffInHours($last_update) : $last;
```

Comment:
The single `=` in `$end = ' hours'` is not a typo here or in the next two lines, is it? That's at pretty peculiar way of doing this. Can't you use the comma operator after the `?` ? Or is that not available in PHP?

Reply:
Indeed it is not a typo.

I tried to think of simpler ways to do this, but multiple if statements made the code a bit long and not nice to read.
Here I am abusing the short-circuit evaluation of PHP. 
https://stackoverflow.com/questions/4216754/php-lazy-boolean-evaluation

if `$last` is `0` then this means that is was not the previous value (` days`) so we know that the `$end` will be ` hours`. I am using the fact that the `=` returns `1` when being executed.

Yes, I know it is smart code... But I could not find a prettier/more elegant way to do it.

Reply:
In C I would've written it like that _(untested code, may need some brackets etc, but you get the gist)_:

```
$last = ($last == 0) ? $end = ' hours', now()->diffInHours($last_update) : $last;
```

No idea if that works in PHP though?